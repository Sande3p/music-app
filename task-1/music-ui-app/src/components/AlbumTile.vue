<template>
  <div class="album-tile">
    <figure>
      <img :src="item.artworkUrl100" :alt="item.collectionName" />
    </figure>
    <h5 class="alubm-artist">{{ item.collectionName }}</h5>
  </div>
</template>

<script>
export default {
  name: "AlbumTile",
  props: ["item"],
  data() {
    return {};
  },
  created() { },
};
</script>
