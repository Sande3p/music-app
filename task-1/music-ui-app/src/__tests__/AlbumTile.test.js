import { mount } from "@vue/test-utils";
import AlbumTile from '../components/AlbumTile.vue';

describe('AlbumTile.test.js', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(AlbumTile, {
      propsData: {
        item: {
          "wrapperType": "collection",
          "collectionType": "Album",
          "artistId": 909253,
          "collectionId": 1469577723,
          "amgArtistId": 468749,
          "artistName": "Jack Johnson",
          "collectionName": "Demo album name",
          "collectionCensoredName": "Demo album name",
          "artworkUrl100": "https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/100x100bb.jpg"
        }
      }
    });
  });

  it("has received 'Demo album name' as the collectionName property", () => {
    expect(cmp.vm.item.collectionName).toEqual('Demo album name')
  })

  it("has received alubm art url", () => {
    expect(cmp.vm.item.artworkUrl100).toEqual('https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/100x100bb.jpg')
  })

  it("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  });
});