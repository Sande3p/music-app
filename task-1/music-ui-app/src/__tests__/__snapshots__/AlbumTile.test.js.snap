// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumTile.test.js has the expected html structure 1`] = `
<div
  class="album-tile"
>
  <figure>
    <img
      alt="Demo album name"
      src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/100x100bb.jpg"
    />
  </figure>
   
  <h5
    class="alubm-artist"
  >
    Demo album name
  </h5>
</div>
`;
